dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -1063,7 +1069,11 @@ Spdp::data_received(const DataSubmessage& data,
     }
     return;
   }
-  if (!is_security_enabled()) {
+
+  const bool security_enabled = is_security_enabled();
+  guard.release();
```

Comment:
This line is conditional on `OPENDDS_SECURITY`.  Was that intended?

Reply:
Nope... will take a look.